### PR TITLE
Firebase

### PR DESCRIPTION
**Pull request by washburnad**
Live refresh of game when other player makes a move
Fixed bug in games controller to prevent error if url is entered directly for non-existant game  e.g (/games/39)

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -109,4 +110,27 @@ $(document).ready(function() {
...
\ No newline at end of file
+
+  var gameID = "game"+ $('#gameboard').data('game-id')
+  var fbRef = new Firebase('https://amber-inferno-5356.firebaseio.com/');
+  var indexRef = fbRef.child(gameID);
+
+  function updateGame(snap, Id) {
+    if (snap.key() == 'game')
+      console.log('refresh game # ',snap.val() );
+      console.log(snap.val());
```
Comment (kenmazaika) [inline app/assets/javascripts/application.js]: console.log breaks certain versions of IE.  These should be removed.

File: app/controllers/pieces_controller.rb [hunk 2]
```diff
@@ -20,4 +21,12 @@ def piece_params
   def your_turn?
     @game.turn == current_player.id
   end
+
+  def update_firebase(data)
+    base_uri = 'https://amber-inferno-5356.firebaseio.com/'
+    firebase = Firebase::Client.new(base_uri)
```
Comment (kenmazaika) [inline app/controllers/pieces_controller.rb]: I'd move this into an initializer.

`config/initializer/firebase.rb`

```
base_uri = 'https://amber-inferno-5356.firebaseio.com/'
FIREBASE = Firebase::Client.new(base_uri)
```

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -17,6 +17,8 @@
 
 
 $(document).ready(function() {
+  var load_time_stamp = Math.floor(Date.now()/1000);
```
Comment (kenmazaika) [inline app/assets/javascripts/application.js]: This is a bit dangerous.  You're using timestamps from two different sources.  One is the server.  The other is the client.  In most cases they should be synced up.  But.  If I'm an idiot and set my date to the wrong date on my computer things will go bad.

I suggest using server time as the definitive source of the time.

Something like this in `application.html.erb`

```
<body data-timestamp="<%= Time.now.to_i %>">
```

**Comment by kenmazaika** [comment]
You guys are coding machines!

**Comment by washburnad** [comment]
Should be ready to go now.

**Pull request merged**